### ISSUE TITLE
Incorrect MarkupHandling inference when batch fixer does not do anything

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn-sdk#498]
This situation appeared in #497.

1. Set up the `TestCode` with input containing markup
2. Set up `FixedCode` with fixed input (no markup or diagnostics)
3. Set up `BatchFixedCode` with `TestCode`

In this case, the assertion behavior is supposed to do the following:

* The iterative fixer works like a call to `VerifyCS.VerifyCodeFixAsync(TestCode, FixedCode)`
* The batch fixer works like a call to `VerifyCS.VerifyCodeFixAsync(TestCode, BatchFixedCode)`

Unfortunately, when a difference is detected between `FixedCode` and `BatchFixedCode`, it automatically selects a `MarkupHandling` default that ignores fixable diagnostics.

📝 Note that when this is fixed, in would be easy to accidentally update the number of expected iterations for the batch fixer to be 0 (which is what happens when `TestCode` and `FixedCode` are the same). However, in this case we need the value to be 1 because the code fix _is offered_, and we only determine that the batch fix has different behavior after that fix is selected.

💡 The easiest way to construct a test case for this is to implement a fix all provider that takes no action, and then provide that fix all provider from a test code fix. The verification call can then be constructed in the same manner as #497, except omit the line that explicitly sets `MarkupHandling`.

```csharp
await new TestWithEmptyFixAllProvider
{
  TestCode = testCode,
  FixedCode = fixedCode,
  BatchFixedCode = testCode,
}.RunAsync();
```
